pennylane/ops/functions/generator.py

```diff
@@ -0,0 +1,175 @@
+        format (str): The format to return the generator in. Must be one of ``'prefactor'``,
+            ``'observable'``, or ``'hamiltonian'``. See below for more details.
+
+    Returns:
+        .Observable or tuple[float, .Observable]: The returned generator, with format/type
+        dependent on the ``format`` argument.
+
+        * ``"prefactor"``: Return the generator as ```(prefactor, obs)`` (representing
+          :math:`G=p \hat{O}`), where
+
+          - prefactor :math:`p` is a float
+          - observable `\hat{O}` is one of :class:`~.Hermitian`,
+            :class:`~.SparseHamiltonian`, or a tensor product
+            of Pauli words.
+
+        * ``"observable"``: Return the generator as a single observable as directly defined
+          by ``op``. Returned generators may be any type of observable, including
+          :class:`~.Hermitian`, :class:`~.Tensor`,
+          :class:`~.SparseHamiltonian`, or :class:`~.Hamiltonian`.
+
+        * ``"hamiltonian"``: Similar to ``"observable"``, however the returned observable
+          will always be converted into :class:`~.Hamiltonian` irregardless of how ``op``
+          encodes the generator.
+
+    **Example**
+
+    Given an operation, ``qml.generator`` returns the generator representation:
+
+    >>> op = qml.CRX(0.6, wires=[0, 1])
+    >>> qml.generator(op)
+    (-0.5, Projector([1], wires=[0]) @ PauliX(wires=[1]))
+
+    It can also be used in a functional form:
+
+    >>> qml.generator(qml.CRX)(0.6, wires=[0, 1])
+    (-0.5, Projector([1], wires=[0]) @ PauliX(wires=[1]))
+
+    By default, ``generator`` will return the generator in the format of ``(prefactor, obs)``,
+    corresponding to :math:`G=p \hat{O}`, where the observable :math:`\hat{O}` will
+    always be given in as in tensor product form, or as a dense/sparse matrix.
+
+    By using the ``format`` argument, the returned generator representation can
+    be altered:
+
+    >>> op = qml.RX(0.2, wires=0)
+    >>> qml.generator(op, format="prefactor")  # output will always be (prefactor, obs)
+    (1.0, Projector([1], wires=[0]))
+    >>> qml.generator(op, format="hamiltonian")  # output will always be a Hamiltonian
+    <Hamiltonian: terms=1, wires=[0]>
+    >>> qml.generator(op, format="observable")  # ouput will be a simplified obs where possible
+    Projector([1], wires=[0])
+    """
+    if op.num_params != 1:
+        raise ValueError(f"Operation {op.name} is not written in terms of a single parameter")
+
+    gen = op.generator()
+
+    if not isinstance(gen, qml.operation.Observable):
+        raise qml.QuantumFunctionError(f"Generator {gen} is not an observable")
+
+    if format == "prefactor":
+        return _generator_prefactor(gen, op)
+
+    if format == "hamiltonian":
+        return _generator_hamiltonian(gen, op)
+
+    if format == "observable":
+        return _generator_observable(gen, op)
+
+    raise ValueError("'format' must be one of ('prefactor', 'hamiltonian', 'observable'")
```

Comment:
```suggestion
    raise ValueError("format must be one of ('prefactor', 'hamiltonian', 'observable')")
```
(Closing bracket was missing in-text and `format` is not a string content but a variable name, so shouldn't have quotation marks)